h/static/scripts/directive/excerpt.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name excerpt
+ * @restrict E
+ * @description This directive truncates its contents to a height specified in
+ *              CSS, and provides controls for expanding and collapsing the
+ *              resulting truncated element. For example, with the following
+ *              template HTML:
+ *
+ *                  <article class="post">
+ *                    <excerpt>
+ *                      <div class="body" ng-model="post.body"></div>
+ *                    </excerpt>
+ *                  </article>
+ *
+ *              You would need to define the allowable height of the excerpt in
+ *              CSS:
+ *
+ *                  article.post .excerpt {
+ *                    max-height: 10em;
+ *                  }
+ *
+ *              And the excerpt directive will take care of the rest.
+ *
+ *              You can selectively disable truncation by providing a boolean
+ *              expression to the `enabled` parameter, e.g.:
+ *
+ *                  <excerpt enabled="!post.inFull">...</excerpt>
+ */
+module.exports = function () {
+  return {
+    link: function (scope, elem, attr, ctrl) {
+      scope.collapsed = true;
+
+      if (typeof scope.enabled === 'undefined') {
+        scope.enabled = function () { return true; };
+      }
+
+      scope.isOverflowing = function () {
```

Comment:
Hmm. It sort of works. Unfortunately resizing the sidebar doesn't currently kick off a digest cycle...